libs/providers/flagsmith/src/lib/flagsmith-provider.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  EvaluationContext,
+  FlagValue,
+  JsonValue,
+  Logger,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderEvents,
+  ProviderMetadata,
+  ResolutionDetails,
+  ResolutionReason,
+} from '@openfeature/web-sdk';
+import { createFlagsmithInstance } from 'flagsmith';
+import { IFlagsmith, IInitConfig, IState } from 'flagsmith/types';
+import { FlagType, typeFactory } from './type-factory';
+
+export class FlagsmithProvider implements Provider {
+  readonly metadata: ProviderMetadata = {
+    name: FlagsmithProvider.name,
+  };
```

Comment:
```suggestion
  };
  
  readonly runsOn = 'client';
```

You can add this to have exceptions thrown if the provider is used by the wrong SDK. cc @beeme1mr